task_processing/plugins/kubernetes/kubernetes_pod_executor.py

```diff
@@ -107,21 +108,175 @@ def _pod_event_watch_loop(self) -> None:
                         "Exception encountered while watching Pod events - restarting watch!")
         logger.debug("Exiting Pod event watcher - stop requested.")
 
+    def __handle_deleted_pod_event(self, event: PodEvent) -> None:
+        pod = event["object"]
+        pod_name = pod.metadata.name
+        task_metadata = self.task_metadata[pod_name]
+
+        logger.info(f"Removing {pod_name} from state and emitting 'killed' event.")
+
+        self.task_metadata = self.task_metadata.discard(pod_name)
+        self.event_queue.put(
+            task_event(
+                task_id=pod_name,
+                terminal=True,
+                success=False,
+                timestamp=time.time(),
+                raw=event["raw_object"],
+                task_config=task_metadata.task_config,
+                platform_type="killed"
+            )
+        )
+
+    def __handle_modified_pod_event(self, event: PodEvent) -> None:
+        pod = event["object"]
+        pod_name = pod.metadata.name
+        task_metadata = self.task_metadata[pod_name]
+
+        if pod.status.phase not in {"Succeeded", "Failed", "Running", "Unknown"}:
```

Comment:
minor-nit: Can this be declared in a variable?
`POD_PHASE = {"Succeeded", "Failed", "Running", "Unknown"}`